### PR TITLE
Feature/add final sql check setting

### PR DESCRIPTION
**Pull request by PavelPancocha**
## Description
Add setting `CACHALOT_FINAL_SQL_CHECK` + create & adapt tests

### CODE REVIEW

File: cachalot/tests/test_utils.py [hunk 1]
```diff
@@ -36,7 +37,9 @@ def force_reopen_connection(self):
     def assert_tables(self, queryset, *tables):
         tables = {table if isinstance(table, str)
                   else table._meta.db_table for table in tables}
-        self.assertSetEqual(_get_tables(queryset.db, queryset.query), tables, str(queryset.query))
+        for setting in (True, False):
```
Comment (dbartenstein) [inline cachalot/tests/test_utils.py]: @Fasther: `final_sql_check` instead of `setting`

Comment (PavelPancocha) [reply cachalot/tests/test_utils.py]: Done :)

File: cachalot/tests/read.py [hunk 2]
```diff
@@ -303,10 +307,20 @@ def test_many_to_many(self):
                 name='Can touch', content_type=ct, codename='touch'),
             Permission.objects.create(
                 name='Can cuddle', content_type=ct, codename='cuddle'))
-        qs = u.user_permissions.values_list('codename', flat=True)
+        return u.user_permissions.values_list('codename', flat=True)
+
+    @with_final_sql_check
+    def test_many_to_many_when_sql_check(self):
```
Comment (PavelPancocha) [inline cachalot/tests/read.py:340]: @dbartenstein please confirm, this is how it should have looked :)

File: cachalot/tests/tests_decorators.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import logging
```
Comment (dbartenstein) [inline cachalot/tests/tests_decorators.py:1]: @Fasther: wouldn’t name file `tests_decorators.py` as it suggests that it contains tests. Better go with `decorators.py`.

File: cachalot/tests/tests_decorators.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def all_final_sql_checks(func):
+    """
+    Runs test as two sub-tests:
+    one with CACHALOT_FINAL_SQL_CHECK setting True, one with False
+    """
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        for i, sql_check in enumerate((True, False)):
```
Comment (dbartenstein) [inline cachalot/tests/tests_decorators.py]: @Fasther: what do you need the `i` and `enumerate` here for? It does not seem to be used.
Better call it `final_sql_check` (not `sql_check`)

File: cachalot/tests/tests_decorators.py (hunk 3 above)
Comment (PavelPancocha) [reply cachalot/tests/tests_decorators.py:1]: @dbartenstein I did it on purpose - all tests are in files without "tests" but the utils live in "tests_utils" where are no tests :)

File: cachalot/tests/tests_decorators.py (hunk 4 above)
Comment (PavelPancocha) [reply cachalot/tests/tests_decorators.py]: left-over :)

**Comment by PavelPancocha** [comment]
@dbartenstein all tests run for me OK.

**Pull request merged**